### DISCUSSION TITLE
How often are application versions released?

### DISCUSSION THREAD

**Discussion by lukasz18dg** [mealie-recipes/mealie#2349] [Q&A]
Q: How often are new versions of the application released?

I ask because I have been using "CopyMeThat" so far and I haven't seen any options for migrating recipes from one system to another, apart from manually copying them from one app to another. Fortunately, someone came up with an idea to implement such migration: https://github.com/hay-kot/mealie/pull/2212
I saw that it was included in the nightly builds, but no official release of the application has been made since then.
And I am concerned about using an application in nightly build on production.

**Comment by hay-kot** [comment]
> How often are new versions of the application released?

We don't have a regular schedule yet, it's something I'd like to setup this year. Currently we're milestone based releases. 

This is the next milestone for the RC-1 release 

https://github.com/hay-kot/mealie/milestone/3

As for a timeline, I can't really give you one, I'm buying a house and moving across the country this summer, so my time is very limited for the next few months.

**Comment by lukasz18dg** [comment]
Thank you for your response.
The last question, before I close this inquiry, is: If I do not use a nightly build, will I have a chance to export from a higher version of the application to a lower version of the application? Or do you see any chance of doing so?
The easiest version, using Docker, to downgrade from the nightly version to beta version does not work. There are some issues with starting the application.

`INFO: 17-Apr-23 21:28:04 	Database connection established.
mealie-api         | INFO: 17-Apr-23 21:28:04 	Context impl PostgresqlImpl.
mealie-api         | INFO: 17-Apr-23 21:28:04 	Will assume transactional DDL.
mealie-api         | INFO: 17-Apr-23 21:28:04 	Migration needed. Performing migration...
mealie-api         | INFO: 17-Apr-23 21:28:04 	Context impl PostgresqlImpl.
mealie-api         | INFO: 17-Apr-23 21:28:04 	Will assume transactional DDL.
mealie-api         | Traceback (most recent call last):
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/script/base.py", line 233, in _catch_revision_errors
mealie-api         |     yield
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/script/base.py", line 443, in _upgrade_revs
mealie-api         |     for script in reversed(list(revs))
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/script/revision.py", line 799, in iterate_revisions
mealie-api         |     revisions, heads = fn(
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/script/revision.py", line 1454, in _collect_upgrade_revisions
mealie-api         |     current_revisions = self.get_revisions(lower)
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/script/revision.py", line 527, in get_revisions
mealie-api         |     return sum([self.get_revisions(id_elem) for id_elem in id_], ())
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/script/revision.py", line 527, in <listcomp>
mealie-api         |     return sum([self.get_revisions(id_elem) for id_elem in id_], ())
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/script/revision.py", line 552, in get_revisions
mealie-api         |     return tuple(
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/script/revision.py", line 553, in <genexpr>
mealie-api         |     self._revision_for_ident(rev_id, branch_label)
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/script/revision.py", line 624, in _revision_for_ident
mealie-api         |     raise ResolutionError(
mealie-api         | alembic.script.revision.ResolutionError: No such revision or branch '38514b39a824'
mealie-api         | 
mealie-api         | The above exception was the direct cause of the following exception:
mealie-api         | 
mealie-api         | Traceback (most recent call last):
mealie-api         |   File "/app/mealie/db/init_db.py", line 107, in <module>
mealie-api         |     main()
mealie-api         |   File "/app/mealie/db/init_db.py", line 93, in main
mealie-api         |     command.upgrade(alembic_cfg, "head")
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/command.py", line 322, in upgrade
mealie-api         |     script.run_env()
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/script/base.py", line 569, in run_env
mealie-api         |     util.load_python_file(self.dir, "env.py")
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/util/pyfiles.py", line 94, in load_python_file
mealie-api         |     module = load_module_py(module_id, path)
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/util/pyfiles.py", line 110, in load_module_py
mealie-api         |     spec.loader.exec_module(module)  # type: ignore
mealie-api         |   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
mealie-api         |   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
mealie-api         |   File "/app/alembic/env.py", line 77, in <module>
mealie-api         |     run_migrations_online()
mealie-api         |   File "/app/alembic/env.py", line 71, in run_migrations_online
mealie-api         |     context.run_migrations()
mealie-api         |   File "<string>", line 8, in run_migrations
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/runtime/environment.py", line 853, in run_migrations
mealie-api         |     self.get_context().run_migrations(**kw)
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/runtime/migration.py", line 611, in run_migrations
mealie-api         |     for step in self._migrations_fn(heads, self):
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/command.py", line 311, in upgrade
mealie-api         |     return script._upgrade_revs(revision, rev)
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/script/base.py", line 431, in _upgrade_revs
mealie-api         |     with self._catch_revision_errors(
mealie-api         |   File "/usr/local/lib/python3.10/contextlib.py", line 153, in __exit__
mealie-api         |     self.gen.throw(typ, value, traceback)
mealie-api         |   File "/opt/pysetup/.venv/lib/python3.10/site-packages/alembic/script/base.py", line 265, in _catch_revision_errors
mealie-api         |     raise util.CommandError(resolution) from re
mealie-api         | alembic.util.exc.CommandError: Can't locate revision identified by '38514b39a824'`

**Reply by hay-kot** [reply]
No, downgrades are not a use case we're testing or planning on supporting.
